lua/zotero/init.lua

```diff
@@ -62,16 +63,14 @@ local function get_attachment_options(item)
   return options
 end
 
-local function open_url(url)
-  local open_cmd
-  if vim.fn.has 'win32' == 1 then
-    open_cmd = 'start'
-  elseif vim.fn.has 'macunix' == 1 then
-    open_cmd = 'open'
-  else -- Assume Unix
-    open_cmd = 'xdg-open'
+local function open_url(url, file_type)
+  if file_type == 'pdf' and M.config.pdf_opener then
+    -- Use the custom PDF opener if specified
+    vim.fn.system(M.config.pdf_opener .. ' ' .. vim.fn.shellescape(url))
+  else
+    -- Use vim.ui.open for other file types or if no custom PDF opener is set
+    vim.ui.open(url)
```

Comment:
Are there really situations in which a user may want to use a pdf program different from their default when opening a pdf via this extension? If not we can just remove the config for this and replace this function entirely with `vim.ui.open`.

Reply:
Mainly, Windows users who use Neovim in WSL2 instead of directly on Windows face this issue. If you specify "okular," it will use the Linux application. However, if you need to use Windows software, you should use "wslopen." This is the only workaround I have found and is not something handled by vim.ui.open.

Currently, there is a problem with vim.ui.open and shellescape in Neovim, as far as I can tell. While there is no issue using vim.ui.open to open PDFs and DOIs, it won't open Zotero entries in Zotero.

Therefore, the open_url function still needs to exist for the purpose of open_in_zotero, if that's functionality you agree is worthy keeping and possibly for filtering PDFs by tags in the future, for instance.